src/MongoDB.Bson/Serialization/BsonMemberMap.cs

```diff
@@ -580,18 +582,14 @@ private static object GetDefaultValue(Type type)
                 throw new BsonSerializationException(message);
             }
 
-            var sourceType = fieldInfo.DeclaringType;
-            var method = new DynamicMethod("Set" + fieldInfo.Name, null, new[] { typeof(object), typeof(object) }, true);
-            var gen = method.GetILGenerator();
-
-            gen.Emit(OpCodes.Ldarg_0);
-            gen.Emit(OpCodes.Castclass, sourceType);
-            gen.Emit(OpCodes.Ldarg_1);
-            gen.Emit(OpCodes.Unbox_Any, fieldInfo.FieldType);
-            gen.Emit(OpCodes.Stfld, fieldInfo);
-            gen.Emit(OpCodes.Ret);
+            var objParameter = Expression.Parameter(typeof(object), "obj");
```

Comment:
It looks like this way is more good looking in general, even if we don't consider that the previous way doesn't work on some platforms

Reply:
Note: I believe that one reason we used Emit was that it allowed us to set the value of a *private* field during deserialization.

If that's true (a test would help determine that) then this is a *breaking* change.